pkg/apis/enterprise/v1alpha3/common_types.go

```diff
@@ -67,3 +67,25 @@ type CommonSplunkSpec struct {
 	// IndexerClusterRef refers to a Splunk Enterprise indexer cluster managed by the operator within Kubernetes
 	IndexerClusterRef corev1.ObjectReference `json:"indexerClusterRef"`
 }
+
+// SmartStoreSpec defines Splunk indexes and remote storage volume configuration
+type SmartStoreSpec struct {
+	// List to accommodate multiple volumes.
+	// For now, a hard check is added such that only one volume is supported to accommodate one common secrete object
+	S3VolList []S3VolumeSpec `json:"volumes,ommitempty"`
+
+	IndexList []IndexSpec `json:"indexes,ommitempty"`
```

Comment:
please comment each public field. I'm also surprised/concerned that "make lint" isn't failing this?

Reply:
Sorry, i missed to reply earlier. No, make lint is not failing.